MonoGame.Framework/Storage/StorageContainer.cs

```diff
@@ -0,0 +1,279 @@
+
+    /// <summary>
+    /// Contains a logical collection of files used for user-data storage.
+    /// </summary>			
+    /// <remarks>MSDN documentation contains related conceptual article: https://learn.microsoft.com/en-us/previous-versions/windows/xna/bb199074(v=xnagamestudio.40)</remarks>
+    partial class StorageContainer : IDisposable
+    {
+        internal readonly string _storagePath;
+        private readonly PlayerIndex? _playerIndex;
+
+        /// <summary>
+        /// Gets a bool value indicating whether the instance has been disposed.
+        /// </summary>
+        public bool IsDisposed { get; private set; }
+
+        private readonly string _displayName;
+        /// <summary>
+        /// Returns display name of the title.
+        /// </summary>
+        public string DisplayName
+        {
+            get { return _displayName; }
+        }
+
+        private readonly StorageDevice _storageDevice;
+        /// <summary>
+        /// Returns the <see cref="StorageDevice"/> that holds logical files for the container.
+        /// </summary>
+		public StorageDevice StorageDevice
+        {
+
+            get { return _storageDevice; }
+        }
+
+        /// <summary>
+        /// Fired when <see cref="Dispose"/> is called or object if finalized or collected by the garbage collector.
+        /// </summary>
+        public event EventHandler<EventArgs> Disposing;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StorageContainer"/> class.
+        /// </summary>
+        /// <param name='device'>The attached storage-device.</param>
+        /// <param name='displayName'> name.</param>
+        /// <param name='playerIndex'>The <see cref="PlayerIndex"/> of the player to save the data.</param>
+        internal StorageContainer(StorageDevice device, string displayName, PlayerIndex? playerIndex)
+        {
+            if (string.IsNullOrEmpty(displayName))
+                throw new ArgumentNullException("A title name has to be provided in parameter name.");
+
+            _storageDevice = device;
+            _displayName = displayName;
+            _playerIndex = playerIndex;
+
+            // TODO Do we want to be consistent and put everything under SavedGames across all platforms?
+#if DESKTOPGL 
+            // We already have a SaveData folder on Desktop.
+            var savedGames = StorageDevice.StorageRoot;
+#else
+            var storageRoot = StorageDevice.StorageRoot;
+            var savedGames = Path.Combine(storageRoot, "SavedGames");
+#endif
+            _storagePath = Path.Combine(savedGames, _displayName);
+
+            // If we have a PlayerIndex use that, otherwise save to AllPlayers folder
+            _storagePath = _playerIndex.HasValue ? Path.Combine(_storagePath, "Player" + (int)_playerIndex.Value) : Path.Combine(_storagePath, "AllPlayers");
+
+            // Create the "device" if need be
+            if (!Directory.Exists(_storagePath))
```

Comment:
All file operations should be moved into the platform specific partial classes as some platforms do not use the file system for save games.